hak/matrix-seshat/build.ts

```diff
@@ -265,12 +270,14 @@ async function buildMatrixSeshat(hakEnv, moduleInfo) {
     // it for now: we should confirm how much of this it still actually needs.
     const env = hakEnv.makeGypEnv();
 
-    if (!hakEnv.isLinux()) {
-        Object.assign(env, {
-            SQLCIPHER_STATIC: 1,
-            SQLCIPHER_LIB_DIR: path.join(moduleInfo.depPrefix, 'lib'),
-            SQLCIPHER_INCLUDE_DIR: path.join(moduleInfo.depPrefix, 'include'),
-        });
+    Object.assign(env, {
+        SQLCIPHER_STATIC: 1,
+        SQLCIPHER_LIB_DIR: path.join(moduleInfo.depPrefix, 'lib'),
+        SQLCIPHER_INCLUDE_DIR: path.join(moduleInfo.depPrefix, 'include'),
+    });
+
+    if (hakEnv.isLinux()) {
+        env.RUSTFLAGS = '-Clink-arg=-Wl,-Bsymbolic -Clink-arg=-Wl,--exclude-libs,ALL';
```

Comment:
any chance this could get an explanatory comment?

Reply:
The goal here is to ensure seshat-node is built in a way that hides all the symbols node doesn't need to load the plugin.

`-Bsymbolic` ensures the plugin does not import another dynamically linked-in SQLite.
`--exclude-libs ALL` ensures the plugin does not export its SQLite to other dynamically linked-in libraries.

Reply:
@heftig https://github.com/vector-im/element-desktop/pull/334/commits/cf254ddd1d4867c56dd0c7bc65c5f3510d4839ff